e2e/util/namespace.go

```diff
@@ -33,13 +45,27 @@ func CreateNamespace(ctx types.Context, cluster types.Cluster, name string) erro
 		}
 
 		log.Debugf("Namespace %q already exist in cluster %q", name, cluster.Name)
-	}
 
-	log.Debugf("Created namespace %q in cluster %q", name, cluster.Name)
+		managed, err := isManagedByRamenE2e(ctx, cluster, ns)
+		if err != nil {
+			return err
+		}
+
+		if !managed {
+			log.Warnf("Namespace %q in cluster %q exists but is not managed by ramen-e2e (missing label %s=%s). "+
+				"This namespace will not be cleaned up at the end of the test.",
```

Comment:
This will be better, moving the warning to the start, and the reason to the end:

    Namespace %q will not be cleaned up after the test: not managed by ramen-e2e (missing lebel %s=%s)